### DISCUSSION TITLE
How can I generate partial return types for queries?

### DISCUSSION THREAD

**Discussion by kylemiller3** [dotansimha/graphql-code-generator#5330] [Q&A] [answered]
Apologies if this is a common issue but I want to be able to generate a type in Typescript for partial fetches. For instance, if I have a Human and it has age, gender and height and my generated query just fetches height I would like a type that reflects that specific query. Not sure how to accomplish type safety without ugly copy and paste of the auto-generated types.

**Comment by ardatan** [comment, marked as answer]
'typescript-operations' plugins generates types for the queries. You can test it on the website;
https://graphql-code-generator.com
Is that what you're looking for?

**Reply by kylemiller3** [reply]
Sort of, but I find my code littered with things like this: 
```
Pick<
            EntitySummary,
            | 'todayGrowth'
            | 'averageDailyGrowth'
            | 'thisWeekAverageGrowth'
            | 'lastUpdated'
            | ......
          >
```
From the example though it looks like I can use fragments to achieve what I want. Thank you.

**Reply by dotansimha** [reply]
You can either use fragments, or add `preResolveTypes: true` to have more readable output.
